File: Path.cpp
#include "Path.h"

void drawText(double, double, std::string);

Path::Path() : Map() {
  distance = 0;
  pathsize = 0;
}

void Path::fillPath(Map m, std::list<int> route) {
  pathsize = route.size();
  int *array = new int[pathsize];
  nodes = new Neighbor[pathsize];
  copy(route.begin(), route.end(), array);
  for (unsigned int i = 0; i < pathsize; ++i) {
    nodes[i] = m[array[i]];
  }
  distance = 0;
  Point a;
  for (unsigned int i = 0; i < pathsize - 1; ++i) {
    a = nodes[i].getLocation();
    distance += a.distance(nodes[i + 1].getLocation());
  }
}

void Path::emptyPath() {
  pathsize = 0;
  if (nodes != 0) {
    delete[] nodes;
  }
  nodes = 0;
  distance = 0;
}

void Path::display() {
  std::cout << "Distance from " << nodes[0].getID() << " to "
            << nodes[pathsize - 1].getID() << ": ";
  std::cout << distance << std::endl;
  std::cout << "Route: ";
  for (unsigned int i = 0; i < pathsize; ++i) {
    std::cout << nodes[i].getID() << " ";
  }
  std::cout << std::endl;
}

int Path::getDistance() { return distance; }

size_t Path::getPathsize() { return pathsize; }

void Path::setDistance(int dist) { distance = dist; }

void Path::setPathsize(unsigned int length) { pathsize = length; }

void Path::draw() {
  Point one, two;
  if (pathsize == 0) {
    ++pathsize;
  }
  for (unsigned int i = 0; i < pathsize - 1; ++i) {
    one = nodes[i].getLocation();
    two = nodes[i + 1].getLocation();
    rectangleDraw(one, two, 3, 0);
  }
}

File: python/path.py
from point import distance
from map import Map
from copy import deepcopy
from helpers import draw_rectangle

class Path(Map):
  def __init__(self):
    self.distance = 0
    self.path_size = 0


  def fillPath(self, m, route):
    self.path_size = len(route)
    nodes = [Neighbor() for _ in range(self.path_size)]
    for i in range(self.path_size):
      nodes[i] = m[route[i]]
    self.distance = 0
    for i in range(self.path_size - 1):
      self.distance += distance(nodes[i].location, nodes[i + 1].location)


  def emptyPath(self):
    self.path_size = 0
    self.distance = 0


  def display(self):
    print("Distance from " , nodes[0].getID() , " to "
              , nodes[self.path_size - 1].getID() , ": ")
    print(self.distance )
    print("Route: ")
    for node in nodes:
      print(node.getID(), end=' ')


  def draw(self):
    if self.path_size is 0:
      self.path_size += 1
    for i in range(self.path_size - 1):
      one = nodes[i].location
      two = nodes[i + 1].location
      draw_rectangle(one, two, 3, 0)
